### ISSUE TITLE
Fast Clustering - Optimisation / Strategy for very large datasets.

### ISSUE THREAD

**Issue by searchsolved** [UKPLab/sentence-transformers#1496]
I made a tool to cluster keywords using the fast clustering algo, but run into issues (CUDA memory errors) when the data sets are large. 

I'm wondering if there is any strategy that could be enabled to get around this. (For example, I thought about batching up the corpus, but the obvious downside is it won't be able to calculate all cluster relationships between the batches).

Would running locally be a solution using CPU and some kind of large page file etc? Really not sure. 

Basically I'm trying to cluster keyword lists, in colab it fails with a CUDA error > 10,000 keywords. I'd like to be able to say, cluster 1,000,000

Any ideas greatly appreciated.
